internal/service/device.go

```diff
@@ -339,5 +339,19 @@ func (h *ServiceHandler) PatchDevice(ctx context.Context, request server.PatchDe
 
 // (PUT /api/v1/devices/{name}/decommission)
 func (h *ServiceHandler) DecommissionDevice(ctx context.Context, request server.DecommissionDeviceRequestObject) (server.DecommissionDeviceResponseObject, error) {
-	return nil, fmt.Errorf("not yet implemented")
+	orgId := store.NullOrgId
+
+	// soft delete the device in the DB
+	err := h.store.Device().SoftDelete(ctx, orgId, request.Name)
```

Comment:
Why delete the device here?

Reply:
Based on today's conversation, we won't soft delete until the device signals that it is done or until the decommissioning process times out, so it won't happen here after all.